[css-animations-2] Add testing coverage for `animation-composition`

graouts opened the pull request:
There is no test coverage for the [`animation-composition`](https://drafts.csswg.org/css-animations-2/#animation-composition) property. At the moment that property is not implemented by any browser engine that I know of, but WebKit plans to implement it soon. It should be straightforward for Gecko and Blink to add support as well considering they have long-established support for Web Animations composite operations and this CSS property uses existing patterns to expose Web Animations features in CSS Animations.

Adding support for this property in WebKit is tracked by [WebKit bug 232086](https://bugs.webkit.org/show_bug.cgi?id=232086).

I believe these changes are exhaustive and comparable to other `animation-` CSS properties for parsing support. I'll note a couple of more specific things:

- `css/css-animations/CSSAnimation-effect.tentative.html`: check that setting `animation-composition` sets the `composite` property on the effect itself,
- `css/css-animations/KeyframeEffect-getKeyframes.tentative.html`: check that `animation-composition` is accounted for when set on keyframes within a `@keyframes` rule to de-duplicate keys, along with the offset and the timing function.

The rest of the changes are pretty standard.

heycam commented on css/css-animations/KeyframeEffect-getKeyframes.tentative.html:
```diff
@@ -47,6 +47,18 @@
 
+@keyframes anim-simple-composite {
+  from { color: rgb(0, 0, 0); animation-composition: replace; }
+  50%  { color: rgb(0, 0, 255);  animation-composition: add; }
```
Nit: one too many spaces before `animation-composition`.

heycam commented on css/css-animations/parsing/animation-computed.html:
```diff
@@ -15,22 +15,23 @@
-test_computed_value("animation", "none", "0s ease 0s 1 normal none running none");
-test_computed_value("animation", "anim", "0s ease 0s 1 normal none running anim");
+// <single-animation-composition> || [ none | <keyframes-name> ]
+test_computed_value("animation", "1s", "1s ease 0s 1 normal none running replace none");
```
Is this the usual way we update WPTs for new longhand components of existing shorthands? This change will make the test (as of right now) fail in all engines. Not sure if we want to go to the extent of something like:

```js
let composition = CSS.supports("animation-composition", "replace") ? " replace": "";
test_computed_value("animation", "1s", `1s ease 0s 1 normal none running${composition} none`);
...
```

heycam commented on css/css-animations/KeyframeEffect-getKeyframes.tentative.html:
```diff
@@ -113,6 +125,28 @@
   to   { margin-top: 20px; margin-right: 20px; margin-bottom: 20px; }
 }
 
+@keyframes anim-merge-offset-and-composite {
```
I think this test makes sense although I couldn't find where in the specs it's defined that animation-composition participates in the keyframe merging behavior like animation-timing-function does.

graouts replied:
I've filed https://github.com/w3c/csswg-drafts/issues/6935.

heycam reviewed (approved):
r=me though see the comment I left in w3c/csswg-drafts#6930 about the shorthand.

graouts commented:
> r=me though see the comment I left in [w3c/csswg-drafts#6930](https://github.com/w3c/csswg-drafts/pull/6930) about the shorthand.

I will update this PR to not include the shorthand stuff and wait for this issue to be resolved.

graouts committed:
Add testing coverage for the animation-composition property.

Note that this property is not part of the "animation" shorthand property per the current spec.

graouts committed:
Rename new tests as `.tentative.html` as suggested by Brian Birtles during review.

graouts commented:
> Looks good to me although I wonder if the tests added in `css/css-animations/parsing/` should be marked as `tentative` like the other tests for CSS Animations Level 2?

Good point, I've done that in 7bac5bcb5bab8cbf2fec6d708fd50fa13f2fb591.
